db/system_auth_keyspace.hh

```diff
@@ -15,10 +15,10 @@
 namespace db {
 
 namespace system_auth_keyspace {
-    enum class version_t: int64_t {
-        v1 = 1,
-        v2 = 2,
-    };
+    using version_t = std::string_view;
+    static constexpr version_t V1 = "";
+    static constexpr version_t V2 = "2";
+
```

Comment:
Why did you decide to change the in-core representation of the auth version? Enums/integers are a bit easier to pass around and you can perform inequality comparisons with them.

Reply:
to avoid conversion from string but as string_view fiasco shows it wasn't a good decision...